common/web/gesture-recognizer/src/test/test.sh

```diff
@@ -0,0 +1,106 @@
+#!/usr/bin/env bash
+#
+# Builds the include script for the current Keyman version.
+#
+
+# Exit on command failure and when using unset variables:
+set -eu
+
+## START STANDARD BUILD SCRIPT INCLUDE
+# adjust relative paths as necessary
+THIS_SCRIPT="$(greadlink -f "${BASH_SOURCE[0]}" 2>/dev/null || readlink -f "${BASH_SOURCE[0]}")"
+. "$(dirname "$THIS_SCRIPT")/../../../../../resources/build/build-utils.sh"
+## END STANDARD BUILD SCRIPT INCLUDE
+
+# This script runs from its own folder
+cd "$(dirname $THIS_SCRIPT)"
+
+display_usage ( ) {
+  echo "Usage: $0 [configure] [clean] [build] [test]"
+  echo "          [--verbose|-v]"
+  echo "       $0 -h|--help"
+  echo
+  echo "  headless               runs all headless unit tests (CI or local)"
+  echo "  browser                runs all browser-based unit tests in CI mode"
+  echo "  all                    runs all unit test sets in CI mode"
+  echo "  ci                     runs selected unit tests via BrowserStack with CI-oriented reporting"
+  echo "  local                  runs selected unit test sets locally with local-oriented reporting"
+  echo "  browser-debug          runs all browser-based unit tests in persistent local mode."
+}
+
+test-headless ( ) {
+  echo "Headless unit tests for this module have not yet been defined."
+}
+
+test-browser ( ) {
+  KARMA_FLAGS=
+  if [[ $# -eq 1  && $1 == "debug" ]]; then
+    KARMA_CONFIG="manual.conf.js"
+    KARMA_FLAGS="--no-single-run"
+  elif [ $REPORT_STYLE == "local" ]; then
+    KARMA_CONFIG="manual.conf.js"
+  else
+    KARMA_CONFIG="CI.conf.js"
+  fi
+
+  npm run karma -- start src/test/auto/browser/$KARMA_CONFIG "$KARMA_FLAGS"
+}
+
+################################ Main script ################################
+
+# Yes, this is a test script, not a build script.  But why not use the new
+# style anyway?
+
+builder_init "headless browser all ci local browser-debug" "$@"
```

Comment:
With #6986, these become `:target`s or `--option`s. `:all` should not be needed because it will be the default if you do `test` without a target. Let's see what this would look like:

```bash
builder_describe \
  "Run unit tests on gesture recognizer" \
  "test" \
  ":headless" ":browser" ":ci" ":local" ":all-local" \
  "--ci" "--local" "--debug,-d"
builder_parse "$@"

if builder_has_option --ci; then
  REPORT_STYLE=ci
elif builder_has_option --local; then
  REPORT_STYLE=local
else
  # Attempt a TC auto-detect
  ...
fi

if builder_has_action headless; then
  test-headless
  builder_report headless success
fi

if builder_has_action browser; then
  if builder_has_option --debug; then
    test-browser debug
  else
    test-browser
  fi
  builder_report browser success
fi 
```

Sample invocations:
```bash
./test.sh :browser
./test.sh --ci :headless
./test.sh test -d
```

* You can and should add descriptions to each parameter per the notes in #6986.
* Re `--debug,-d` -- I think that this should become a standard option (not necessarily always available), and I need to think of a good way to track these (I will add to the spec doc for now). See how I have added `-d` as a shorthand but always test `--debug` in the code.
* Probably also `--ci` should be a standard option.
* We should always pass `--ci` from TC scripts so `--local` flag should not be needed

For sure I don't want us to be working directly with variables like `builder_chosen_actions` because those are internal to the builder functions :grin: